src/xref2/env.ml

```diff
@@ -235,18 +239,30 @@ let add_label identifier heading env =
   assert env.linking;
   let comp = `Label (identifier, heading) in
   let name = Identifier.name identifier in
-  let () =
+  let ambiguous_labels =
     match ElementsById.find_by_id identifier env.ids with
-    | Some _ ->
-        Format.eprintf "Duplicate label found: %a\n%!"
-          Component.Fmt.model_identifier
-          (identifier :> Identifier.t)
-    | None -> ()
+    | Some (#Component.Element.label as l) ->
+        let err =
+          try
+            match
+              Identifier.Maps.Label.find identifier env.ambiguous_labels
+            with
+            | `Ambiguous (x, others) -> `Ambiguous (x, comp :: others)
+          with Not_found -> `Ambiguous (l, [ comp ])
+        in
+
+        Identifier.Maps.Label.add identifier err env.ambiguous_labels
+    | Some _ -> assert false
```

Comment:
To avoid this assert, we could use `lookup_by_id s_label`, which also return useful data in the ambiguous case.

Reply:
I think this assert is in place of what I actually want, which is a type error. You're right, we could avoid this assert with `s_label`, but `s_label` just does:
```
(function #Component.Element.label as r -> Some r | _ -> None)
```
so I could get the same effect by having just `Some _ -> env.ambiguous_labels`. I don't like we've got a runtime instead of a compile time error here, but I like ignoring the error entirely even less :-)

Reply:
Agree.